### PR TITLE
[base-node] Add rewind-blockchain and improve header-stats command

### PR DESCRIPTION
**Pull request by sdbondi**
## Description

- `rewind-blockchain [to-height]` rewinds the blockchain to the given
  height
- header-stats write to a csv file (custom filename be specified)
- header-stats can filter out monero and sha3 PoW
- fix header-stats command docs

## Motivation and Context

Better blockchain db visibility and debugging info 

## How Has This Been Tested?

Tested on base node

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [ ] I'm merging against the `development` branch.
* [ ] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by delta1** [comment]
Nice! Random cargo fmt issue. Will test :+1:

**Comment by delta1** [comment]
```
>> header-stats
Error: No args provided
```

So I know `help header-stats` shows the message but I think it would be nice to see the help message if no args are provided

File: applications/tari_base_node/src/command_handler.rs [hunk 1]
```diff
@@ -884,66 +888,104 @@ impl CommandHandler {
...
+        pow_algo: Option<PowAlgorithm>,
+    )
+    {
+        let db = self.blockchain_db.clone();
         self.executor.spawn(async move {
-            let mut results: Vec<(u64, usize, u64, u64, u64, PowAlgorithm)> = Vec::new();
+            let mut output = try_or_print!(File::create(&filename));
+
+            println!(
+                "Loading header from height {} to {} and dumping to file [working-dir]/{}.",
```
Comment (delta1) [inline applications/tari_base_node/src/command_handler.rs]: Would be good to have `pow_algo` in this message too

**Review by delta1 — DISMISSED** [review]
LGTM

File: applications/tari_base_node/src/command_handler.rs (hunk 1 above)
Comment (sdbondi) [reply applications/tari_base_node/src/command_handler.rs]: Done :)

**Review by delta1 — APPROVED** [review]

**Comment by delta1** [comment]
Flaky test passed for me locally :+1:

**Review by stringhandler — APPROVED** [review]

**Pull request merged**